### ISSUE TITLE
Credit card prompt does not disappear when no longer selecting a credit card field

### ISSUE THREAD

**Issue by jonalmeida** [mozilla-mobile/fenix#19707]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have the exact bug and steps to reproduce described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

## Steps to reproduce
1. Have a credit card saved on your device.
1. Go to a website that has a credit card input form (example: https://greenido.github.io/Product-Site-101/form-cc-example.html)
2. Click on a credit card field, observe the cc prompt show up below the URL bar.
3. Click anywhere else, like another non-cc field.

### Expected behavior
- The credit card prompt should disappear.

### Actual behavior
- The prompt is always there until you refresh the page.

### Device information

* Device vendor / model and Android version: Android 12, Pixel 3
* Firefox for Android version: Nightly 27/05
 
<img src="https://user-images.githubusercontent.com/1370580/120035768-16ffdd00-bfcd-11eb-894e-2f2bb407aa6f.png" width=320 />

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-10817)

**Comment by VitalyVPinchuk** [comment]
Hi @jonalmeida 
Looks like the issue is in `mozilla.components.feature.prompts`
`ifAnyChanged()` in `PromptFeature.start()` doesn't register any changes when tapping non-cc field so `creditCardPicker?.dismissSelectCreditCardRequest()` doesn't get invoked

```
        handlePromptScope = store.flowScoped { flow ->
            flow.map { state -> state.findTabOrCustomTabOrSelectedTab(customTabId) }
                .ifAnyChanged {
                    arrayOf(it?.content?.promptRequest, it?.content?.loading)
                }
```

**Comment by Mugurell** [comment]
Created https://bugzilla.mozilla.org/show_bug.cgi?id=1715572 for GV to expose a new api based on which we'll know when to cancel this and the SelectLogin prompt.

**Comment by Mugurell** [comment]
This will probably be fixed by incorporating the changes from https://github.com/mozilla-mobile/android-components/issues/10867

**Comment by codrut-topliceanu** [comment]
Waiting on https://bugzilla.mozilla.org/show_bug.cgi?id=1715572 in order to fix this in AC.

**Comment by Mugurell** [comment]
Waiting for GV.

**Comment by Mugurell** [comment]
https://github.com/mozilla-mobile/fenix/issues/19707 landed.
Curious whether this was fixed entirely from GeckoView.

**Comment by boek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1814142

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
